version: '3'
services:
  webdav:
    image: s17n/webdav:$WEBDAV_VERSION
    container_name: webdav
    restart: "unless-stopped"
    networks: 
      - webdav
      - reverse-proxy
    ports:
      - "8008:80" 
    environment:
      auth_type: Digest
    volumes:
      - './data/dav:/var/lib/dav'
      - './secrets/.htpasswd:/user.passwd'
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=reverse-proxy"
      - "traefik.http.routers.devonthink.entrypoints=https"
      - "traefik.http.routers.devonthink.rule=Host(`devonthink.${DOMAIN_NAME}`)"
      - "traefik.http.routers.devonthink.tls=true"
      - "traefik.http.routers.devonthink.tls.certresolver=dns-cloudflare"
      - "traefik.http.services.devonthink.loadbalancer.server.scheme=http"
      - "traefik.http.services.devonthink.loadbalancer.server.port=80"
    logging:
      driver: loki
      options:
        loki-url: $LOKI_URL
        no-file: "true"

networks:
  webdav:
    name: webdav
    driver: bridge
  reverse-proxy:
    external: true